### PR TITLE
 #162949230 Share articles via email and social platforms

### PR DESCRIPTION
**Pull request by sylviawanjiku**
#### What does this PR do?

Implements sharing of articles via email and social sites(Facebook and Twitter)
#### Description of Task to be completed?

- Implement share article via email
- Implement share article facebook and twitter
- Add tests on share articles

#### How should this be manually tested?

1. Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
2. Cd into the cloned repo 
3. Git checkout to branch `ft-share-articles-channels-162949230`
4. Set up a virtual environment and install requirements:
      ```
            python3 -m venv venv
            source venv/bin/activate
            pip3 install -r requirements.txt

     ```
5. Set up a .env file on the same level as the venv file and add the following:
      ```
            source venv/bin/activate
            export SECRET_KEY='yoursecretkey'
             export DATABASE_URL='postgres://yourdatabasename'
             export HOST_PASSWORD=" "
             export HOST_USER=" "
             export SMTP_HOST=' '
             export EMAIL_PORT='587'
             export FACEBOOK_KEY=' '
             export FACEBOOK_SECRET=' '
             export TWITTER_KEY=' '
             export TWITTER_SECRET_KEY=' '
             export GOOGLE_KEY=' '
             export GOOGLE_SECRET=' '
             export ACCESS_TOKEN=' '
             export SECRET_TOKEN=' '
             export BASE_URL="  "


      ```
6. Then run: 
     ``` 
            source .env 
            ./manage.py migrate
            ./manage.py runserver
      ```

7. Fire up postman and test the following endpoints:

| EndPoint                       | Functionality                           |
| ---------------------- |:----------------------------:|
| POST     api/articles/ `<slug>`/email/share  | Share via email                      |
| POST     api/articles/`<slug>`/facebook/share                    |Share via Facebook |
| POST     api/articles/`<slug>`/twitter/share   |Share via Twitter                      |

                                                    
use the following payload format for sharing an article via email:
```
{
    "email": " "
  }
```

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[162949230](https://www.pivotaltracker.com/story/show/162949230)
#### Screenshots (if appropriate)
image

image

#### Questions:

### CODE REVIEW

File: authors/apps/articles/templates/share_article.html [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  <style type="text/css">
+  /**
+   * Google webfonts. Recommended to include the .woff version for cross-client compatibility.
+   */
+  @media screen {
+    @font-face {
+      font-family: 'Source Sans Pro';
+      font-style: normal;
+      font-weight: 400;
+      src: local('Source Sans Pro Regular'), local('SourceSansPro-Regular'), url(https://fonts.gstatic.com/s/sourcesanspro/v10/ODelI1aHBYDBqgeIAH2zlBM0YzuT7MdOe03otPbuUS0.woff) format('woff');
```
Comment (kwanj-k) [inline authors/apps/articles/templates/share_article.html]: Line too long, probably do a line break now and then when the line is too long

**Review by kwanj-k — COMMENTED** [review]
Please update your PR description, it's directing me to a different branch

File: authors/apps/articles/views/articles.py [hunk 2]
```diff
@@ -12,13 +14,21 @@
 # local imports
 from authors.apps.articles.models import Article, LikeDislike
 from authors.apps.articles.serializers import (
-    ArticleSerializer, UpdateArticleSerializer, LikeDislikeSerializer
+    ArticleSerializer, UpdateArticleSerializer,
+    LikeDislikeSerializer,
+    EmailCheckSerializer
```
Comment (indungu) [inline authors/apps/articles/views/articles.py]: Move this to line 18 please

**Review by indungu — CHANGES_REQUESTED** [review]
Please make the following changes.
 
And update the PR description as @kwanj-k requested while you are at it too please.
```markdown
[#162949230](https://www.pivotaltracker.com/story/show/162949230)
```

**Review by indungu — DISMISSED** [review]

File: authors/apps/articles/migrations/0002_auto_20190130_1803.py [hunk 3]
```diff
@@ -15,4 +15,4 @@ class Migration(migrations.Migration):
             name='updated_at',
             field=models.DateTimeField(auto_now=True),
         ),
-    ]
+    ]
```
Comment (kwanj-k) [inline authors/apps/articles/migrations/0002_auto_20190130_1803.py]: Add a new line here

File: authors/apps/articles/migrations/0003_rating.py [hunk 4]
```diff
@@ -22,4 +22,4 @@ class Migration(migrations.Migration):
                 ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
             ],
         ),
-    ]
+    ]
```
Comment (kwanj-k) [inline authors/apps/articles/migrations/0003_rating.py]: Add a new line here

File: authors/apps/articles/templates/share_article.html [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+  /**
+   * Google webfonts. Recommended to include the .woff version for cross-client compatibility.
+   */
+  @media screen {
+    @font-face {
+      font-family: 'Source Sans Pro';
+      font-style: normal;
+      font-weight: 400;
+      src: local('Source Sans Pro Regular'), local('SourceSansPro-Regular'),
+      url(https://fonts.gstatic.com/s/sourcesanspro/v10/ODelI1aHBYDBqgeIAH2zlBM0YzuT7MdOe03otPbuUS0.woff) format('woff');
```
Comment (kwanj-k) [inline authors/apps/articles/templates/share_article.html:18]: Line too long

File: authors/apps/articles/templates/share_article.html [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    }
+    @font-face {
+      font-family: 'Source Sans Pro';
+      font-style: normal;
+      font-weight: 700;
+      src: local('Source Sans Pro Bold'), local('SourceSansPro-Bold'), 
+      url(https://fonts.gstatic.com/s/sourcesanspro/v10/toadOcfmlt9b38dHJxOBGFkQc6VGVFSmCnC_l7QZG60.woff) format('woff');
```
Comment (kwanj-k) [inline authors/apps/articles/templates/share_article.html:25]: Line too long, please consider making it shorter

File: authors/tests/base_test.py [hunk 7]
```diff
@@ -8,6 +8,7 @@
 # local imports
 from authors.apps.authentication.activate import account_activation_token
 from authors.apps.authentication.models import User
+import os
```
Comment (kwanj-k) [inline authors/tests/base_test.py]: I read somewhere such an import should at the top of the file, ie os imports come first, the third party then local imports
And you have put it under the comment 'local imports', kindly look in to that

File: authors/apps/articles/serializers.py [hunk 8]
```diff
@@ -187,3 +187,17 @@ class CommentHistorySerializer(serializers.ModelSerializer):
     class Meta:
         model = Comments
         fields = ('id', 'body', 'created_at', 'updated_at')
+
+
+class EmailCheckSerializer(serializers.Serializer):
```
Comment (Allan690) [inline authors/apps/articles/serializers.py]: This serializer already exists in authentications/serializers.py. I think you can just call that one if you need it in your views. That will help us keep our code DRY.

**Review by kwanj-k — APPROVED** [review]

**Review by nesh-dev — APPROVED** [review]

**Pull request merged**